src/stdlib_experimental_stats.fypp

```diff
@@ -12,92 +10,113 @@ module stdlib_experimental_stats
   public :: mean
 
   interface mean
-
-    #:for k1, t1 in REAL_KINDS_TYPES
+#:def name(Rank, Type, Kind)
+$:"mean_{0}_all_{1}{2}_{1}{2}".format(Rank, Type[0], Kind)
+#:enddef
```

Comment:
I like idea to use a preprocessing macro for the names of functions, especially if we will other functions with the same API (e.g., `variance`, `median`,...)

So, to make these macros more general, would it be possible to pass the name of the function, e.g.,:
`#:def name(Varname, Rank, Type, Kind)`
where `Varname` would be equal to `mean`, `variance`, `median`,....?

Also, would it be possible to add these macros in `common.fypp`, since these will be used for other functions?

Reply:
Yes, I'll do. I started that way but because the original routine names were complex, I decided for hard-code the names in order not to change too much the code. I'll amend and send a new request.

Also @jvdp1, could we make the functions pure? (I've not used much of submodules before, so I don't know if there is restrictions there)

Reply:
Functions in the submodule can be pure. However, it seems that the minimum required CMake does not support that (I am not sure if the last version of CMake supports it either; @zbeekma @certik @milancurcic ?).
Also the checks on the dimensions inside the functions may prohibit `pure` for some of them.